src/hotspot/share/gc/g1/g1CodeRootSetTable.hpp

```diff
@@ -24,44 +24,42 @@
 #ifndef SHARE_GC_G1_G1CODEROOTSETTABLE_HPP
 #define SHARE_GC_G1_G1CODEROOTSETTABLE_HPP
 
-#include "utilities/hashtable.hpp"
+#include "code/codeCache.hpp"
+#include "memory/allocation.hpp"
+#include "utilities/resizeableResourceHash.hpp"
+#include "utilities/resourceHash.hpp"
 
 class nmethod;
 
-class G1CodeRootSetTable : public Hashtable<nmethod*, mtGC> {
+class G1CodeRootSetTable : public CHeapObj<mtGC>  {
   friend class G1CodeRootSetTest;
-  typedef HashtableEntry<nmethod*, mtGC> Entry;
 
   static G1CodeRootSetTable* volatile _purge_list;
 
+  using Table = ResizeableResourceHashtable<nmethod*, nmethod*, AnyObj::C_HEAP, mtGC>;
+  Table _table;
   G1CodeRootSetTable* _purge_next;
 
   unsigned int compute_hash(nmethod* nm) {
```

Comment:
Change makes this dead code

Reply:
I do not understand, I am still "using"  Table in `mem_size`, or did you mean something else?

Reply:
`unsigned int compute_hash(nmethod* nm) {` function,  sorry for not being clear

Reply:
Yes, will fix, nice find!